tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -859,23 +859,23 @@ func (az *AzureProvider) constructNetworkProfile(ctx context.Context,
 		}
 	}
 
-	var vmNic network.Interface
+	var nicID *string
```

Comment:
I would remove this variable and use `nic.ID` directly. But add a check to ensure `nic.ID` is not `nil`